include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,223 @@
+  /// The device address when A0 is connected to SDA.
+  static constexpr hal::byte address_sda = 0b100'1010;
+  /// The device address when A0 is connected to SCL.
+  static constexpr hal::byte address_scl = 0b100'1011;
+
+  /// The address of the read-only register containing the temperature data.
+  static constexpr hal::byte xyz_register = 0x3B;
+  /// The address of the register used to configure gravity scale the device.
+  static constexpr hal::byte configuration_register = 0x1C;
+  /// The address of the register used to initilize the device.
+  static constexpr hal::byte initalizing_register = 0x6B;
+
+  /// The command to enable one-shot shutdown mode.
+  static constexpr hal::byte who_am_i_register = 0x75;
+  
+  /**
+   * @brief Constructs and returns MPU object
+   * 
+   * @param i2c I2C bus the MPU is connected to 
+   * @param device_address 0x68 if address pin is pulled low, 0x69 if pulled high
+   * @return result<mpu6050> 
+   */
+  static result<mpu6050> create(hal::i2c& i2c,
+                                hal::byte device_address = address_ground)
+  {
+    mpu6050 mpu(i2c, device_address);
+    mpu.power_on();
+    return mpu;
+  }
+
+  /**
+   * @brief Changes the gravity scale that the MPU is reading. The larger the
+   * scale, the less precise the reading.
+   *
+   * @param gravity_scale Scales in powers of 2 up to 16.
+   * @return hal::status
+   */
+  hal::status configure_full_scale(uint32_t gravity_scale)
+  {
+    hal::byte gravity_code;
+
+    switch (gravity_scale) {
+      case 2:
+        gravity_code = 0x00;
+        m_gscale = 2.0f;
+        break;
+      case 4:
+        gravity_code = 0x01;
+        m_gscale = 4.0f;
+        break;
+      case 8:
+        gravity_code = 0x02;
+        m_gscale = 8.0f;
+        break;
+      case 16:
+        gravity_code = 0x03;
+        m_gscale = 16.0f;
+        break;
+      default:
+        return hal::new_error(std::errc::illegal_byte_sequence);
+    }
+
+    // Write in the full scale range; but leave the self test and high pass
+    // filter config untouched
+    constexpr auto scale_mask = hal::bit::mask::from<3, 4>();
+
+    auto config =
+      HAL_CHECK(hal::write_then_read<1>(*m_i2c,
+                                        m_address,
+                                        std::array{ configuration_register },
+                                        hal::never_timeout()));
+
+    hal::bit::modify(config[0]).insert<scale_mask>(gravity_code);
+    return hal::write(*m_i2c,
+                      m_address,
+                      std::array{ configuration_register, config[0] },
+                      hal::never_timeout());
+  }
+
+  /**
+   * @brief Re-enables acceleration readings from the MPU
+   *
+   * @return hal::status
+   */
+  hal::status power_on()
+  {
+    HAL_CHECK(is_valid_device());
+    return active_mode(true);
+  }
+
+  /**
+   * @brief Disables acceleration reading from the MPU.
+   *
+   * @return hal::status
```

Comment:
```suggestion
   * @return hal::status - success or errors from i2c communication
```

Add a quick explanation of what the return value could return.